### PR TITLE
eth2: catalyst RPC calls and command line options

### PR DESCRIPTION
**Pull request by gballet**
Supersedes #22320

### CODE REVIEW

File: core/vm/opcodes.go [hunk 1]
```diff
@@ -101,24 +101,29 @@ const (
...
+	MSTORE8   OpCode = 0x53
+	SLOAD     OpCode = 0x54
+	SSTORE    OpCode = 0x55
+	JUMP      OpCode = 0x56
+	JUMPI     OpCode = 0x57
+	PC        OpCode = 0x58
+	MSIZE     OpCode = 0x59
+	GAS       OpCode = 0x5a
+	JUMPDEST  OpCode = 0x5b
+	BEGINSUB  OpCode = 0x5c
```
Comment (MariusVanDerWijden) [inline core/vm/opcodes.go:124]: Not needed anymore

Comment (gballet) [reply core/vm/opcodes.go]: Thanks, fixed

File: eth/eth2.go [hunk 2]
```diff
@@ -0,0 +1,320 @@
...
+
+		default:
+			// Strange error, discard the transaction and get the next in line (note, the
+			// nonce-too-high clause will prevent us from executing in vain).
+			log.Debug("Transaction failed, account skipped", "hash", tx.Hash(), "err", err)
+			txs.Shift()
+		}
+	}
+
+	block, err := api.eth.Engine().FinalizeAndAssemble(bc, header, api.env.state, transactions, nil /* uncles */, api.env.receipts)
```
Comment (rjl493456442) [inline eth/eth2.go:204]: QQ: the block reward is applied based on the ethash rules. So in the eth2, actually there is no block reward here right? The reward will be applied in the consensus layer(they have their own accounting model).

Comment (gballet) [reply eth/eth2.go:204]: Yes, this is the long term view. Right after the merge, you won't be able to move the "eth2 eth" from the beacon chain into the eth1 "execution layer", so it still makes sense to offer a reward... more importantly, for the upcoming testnet: if you don't have a way to produce value, you won't be able to test anything.

File: core/vm/jump_table.go [hunk 3]
```diff
@@ -57,11 +57,30 @@ var (
 	constantinopleInstructionSet   = newConstantinopleInstructionSet()
 	istanbulInstructionSet         = newIstanbulInstructionSet()
 	berlinInstructionSet           = newBerlinInstructionSet()
+	catalystInstructionSet         = newCatalystInstructionSet()
 )
 
 // JumpTable contains the EVM opcodes supported at a given fork.
 type JumpTable [256]*operation
 
+//nolint:deadcode
```
Comment (gballet) [inline core/vm/jump_table.go:66]: node to self: this is no longer needed

**Comment by gballet** [comment]
Closing to prevent confusion with #22641

**Pull request closed without merging**